integration-templates/algolia/algolia-create-contacts.ts

```diff
@@ -0,0 +1,33 @@
+import type { NangoAction, AlgoliaContact, AlgoliaCreateContactInput } from './models';
+
+export default async function runAction(nango: NangoAction, input: AlgoliaCreateContactInput): Promise<AlgoliaContact> {
+    if (!input.name) {
+        throw new nango.ActionError({
+            message: 'name is a required field'
+        });
+    }
+
+    const endpoint = `/1/indexes/contacts`;
+
+    try {
+        const postData = {
+            name: input.name,
+            company: input.company,
+            email: input.email
+        };
+        const resp = await nango.post({
+            endpoint: endpoint,
+            data: postData
+        });
+
+        return {
+            createdAt: resp.data.createdAt,
+            taskID: resp.data.taskID,
```

Comment:
note that the API gives back a TaskID because the indexing is not synchronous. I guess we don't care for the template but it could be something users should be aware of.